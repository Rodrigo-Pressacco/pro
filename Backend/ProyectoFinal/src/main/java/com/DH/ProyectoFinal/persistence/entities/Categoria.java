package com.DH.ProyectoFinal.persistence.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Entity
@Getter
@Setter
@Table(name ="categorias", schema = "neotravel")
public class Categoria {

    @Id
    //@SequenceGenerator(name = "categorias_sequence", sequenceName = "categorias_sequence", allocationSize = 1)
    @GeneratedValue(strategy = GenerationType.IDENTITY /*, generator = "categorias_sequence"*/)
    @Column
    private Integer id;
    @Column
    private String titulo;
    @Column
    private String descripcion;
    @Column
    private String url_imagen;

    @OneToMany(mappedBy = "categoria", fetch = FetchType.EAGER)
    @JsonIgnore
    private Set<Producto> productos = new HashSet<>();

    public Categoria() {
    }

    public Categoria(String titulo, String descripcion, String url_imagen) {
        this.titulo = titulo;
        this.descripcion = descripcion;
        this.url_imagen = url_imagen;
    }

    @Override
    public String toString() {
        return "Categoria{" +
                "id=" + id +
                ", titulo='" + titulo + '\'' +
                ", descripcion='" + descripcion + '\'' +
                ", imagen='" + url_imagen + '\'' +
                '}';
    }
}
