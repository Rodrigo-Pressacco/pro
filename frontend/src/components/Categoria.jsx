import React from 'react';
import "../styles/Categorias.css";

const Categoria = ({nombre, image, cant}) => {


    return (
        
        <div>
            <div className="categoria-body">
                <img src={image} alt="Categoria" />
                <h2>{nombre}</h2>
                <p>{cant}</p>

            </div>
        </div>
    )
}

export default Categoria;

