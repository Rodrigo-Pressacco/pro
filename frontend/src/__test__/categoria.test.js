import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import Categoria from '../components/Categoria'

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';
Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('<Categoria />', () => {
    let categoria
    
    beforeEach(() => {
        categoria = shallow(<Categoria />)
    });

    it("Deberia mostrar <Categoria/> correctamente", () => {
        expect(categoria).toMatchSnapshot();
    });

})