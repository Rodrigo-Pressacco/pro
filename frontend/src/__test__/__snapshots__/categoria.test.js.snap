// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categoria /> Deberia mostrar <Categoria/> correctamente 1`] = `
<div>
  <div
    className="categoria-body"
  >
    <img
      alt="Categoria"
    />
    <h2 />
    <p />
  </div>
</div>
`;
